pkg/agent/openflow/pipeline.go

```diff
@@ -725,8 +740,8 @@ func (c *client) traceflowL2ForwardOutputFlow(dataplaneTag uint8, category cooki
 
 // l2ForwardOutputServiceHairpinFlow uses in_port action for Service
 // hairpin packets to avoid packets from being dropped by OVS.
-func (c *client) l2ForwardOutputServiceHairpinFlow() binding.Flow {
-	return c.pipeline[L2ForwardingOutTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+func (c *client) l2ForwardOutputServiceHairpinFlow(ipProtocol binding.Protocol) binding.Flow {
```

Comment:
As this flow doesn't match any IP or ct_mark, I am thinking is it possible to remove the match condition of "ip" or "ipv6", then we could trate the packet as a L2 packet, and only one flow entry could satisfy the requirement.

Reply:
Wouldn't that affect other protocols e.g arp? IDK if that has any importance as I think that these packets are treated earlier in the chain.

Reply:
I don't think so. As ARP packets would be forwarded in table20, it can't go ino this table.

Reply:
Make sense. I'll apply this change. Thanks!